src/main/java/FreedomPointsCounter.java

```
@@ -0,0 +1,120 @@
+        for (Map.Entry<Position, Cell> entry : board) {
+            Position currentPosition = entry.getKey();
+            checkFreedomLineFrom(currentPosition);
+        }
+    }
+
+    private void checkFreedomLineFrom(@NotNull Position position) {
+
+        Stone stone = board.getStone(position);
+        if(stone == null) return;
+        Stone.Color playerColor = stone.getColor();
+
+        int currentStoneCount = 1;
+        if (countStonesOfTheSameColorFrom(position, currentStoneCount, Direction.HORIZONTAL) == MAX_NUMBER_OF_STONES) {
+            incrementScoreOf(playerColor);
+        } else if (countStonesOfTheSameColorFrom(position, currentStoneCount, Direction.VERTICAL) == MAX_NUMBER_OF_STONES) {
+            incrementScoreOf(playerColor);
+        } else if (countStonesOfTheSameColorFrom(position, currentStoneCount, Direction.DIAGONAL_LEFT) == MAX_NUMBER_OF_STONES) {
+            incrementScoreOf(playerColor);
+        } else if (countStonesOfTheSameColorFrom(position, currentStoneCount, Direction.DIAGONAL_RIGHT) == MAX_NUMBER_OF_STONES) {
+            incrementScoreOf(playerColor);
+        }
+
+    }
+
+    private void incrementScoreOf(@NotNull Stone.Color playerColor) {
+        if (playerColor == Stone.Color.WHITE) {
+            whitePlayerScore = whitePlayerScore + 1;
+        } else {
+            blackPlayerScore = blackPlayerScore + 1;
+        }
+    }
+
+    private int countStonesOfTheSameColorFrom(@NotNull Position currentPosition, int currentStoneCount, @NotNull Direction direction) {
+        Position nextPosition;
+        if (currentStoneCount == 1) {
+            try {
+                Stone previousStone = getThePreviousStone(currentPosition, direction);
+                if (previousStone != null && previousStone.getColor() == board.getStone(currentPosition).getColor()) {
+                    return currentStoneCount;
+                }
+            } catch (InvalidPositionException ignored) {
+
```

In the next iteration, the ignored exceptions should at least produce some logging information